### DISCUSSION TITLE
register_external_context_pane shouldn't change context.layout

### DISCUSSION THREAD

**Discussion by MrQubo** [hugsy/gef#1015] [General]
`register_external_context_pane()` automatically appends pane name to `context.layout`.

I think that `register_external_context_pane()` shouldn't modify `context.layout`. Users can set the layout themselves or the code calling register_external_context_pane could do so.

Just remove this line: https://github.com/hugsy/gef/blob/5927df4fb307124c444453b1cb85fa0ce79883c9/gef.py#L9607

Seems like issue template thing is broken for me. Clicking "Submit new issue" just does nothing.

**Comment by hugsy** [comment]
> `register_external_context_pane()` automatically appends pane name to `context.layout`.

It's debatable: I'd argue that if you go to the point of writing your own pane(s) it means you intend to use them so automatically adding them makes sense.

Did this break something for you (aside from the new layout appearance)?

> Seems like issue template thing is broken for me. Clicking "Submit new issue" just does nothing.

Weird 🤔

**Comment by hugsy** [comment]
Also, moving this into discussion, as it's not a bug

**Comment by MrQubo** [comment]
Based on the name of the function, which says "register" I think it makes more sense to only register and not add to the layout.
This behaviour could be split into another method, e.g. `add_external_context_pane()`, or hidden behind an optional kwarg argument.
But this would break backward compatibilty only for a minor naming issue.

**Comment by MrQubo** [comment]
For my use case.
I have already set this pane in `context.layout` and this results in it being duplicated and in the wrong position.
An alternate fix that wouldn't affect existing behaviour would be to check if the pane is already in layout and add it only if it isn't.

**Comment by therealdreg** [comment]
solved for syscall-args glibc-args 

@MrQubo , look this! #1093 and https://github.com/hugsy/gef-extras/pull/111 

https://hugsy.github.io/gef/api/#context-layout-mapping-api

So, @MrQubo Can I close this discussion?

If you need to check on each execution whether it's added or not... a simple hack could be:

```
pi gef.config['context.layout'] += ' your_plug' if 'your_plug' not in gef.config['context.layout'] else ''
``` 

- https://github.com/hugsy/gef-extras/archive/refs/heads/main.zip
- https://github.com/hugsy/gef/archive/refs/heads/main.zip

**Comment by therealdreg** [comment]
solved #1117
